sdk/typescript/src/utils/sui-types.ts

```diff
@@ -108,3 +122,51 @@ function isHex(value: string): boolean {
 function getHexByteLength(value: string): number {
 	return /^(0x|0X)/.test(value) ? (value.length - 2) / 2 : value.length / 2;
 }
+
+/**
+ * This returns a Keypair object based by validating the 33-byte Bech32 encoded string
+ * starting with `suiprivkey`, and construct the keypair based on the signature scheme
+ * and the private key bytes.
+ */
+export function decodeSuiPrivateKey(value: string): Keypair {
+	const { prefix, words } = bech32.decode(value);
+	if (prefix !== SUI_PRIVATE_KEY_PREFIX) {
+		throw new Error('invalid private key key');
+	}
+	const extendedSecretKey = new Uint8Array(bech32.fromWords(words));
+	let secretKey = extendedSecretKey.slice(1);
+	const signatureScheme =
+		SIGNATURE_FLAG_TO_SCHEME[extendedSecretKey[0] as keyof typeof SIGNATURE_FLAG_TO_SCHEME];
+
+	switch (signatureScheme) {
+		case 'ED25519':
+			let pureSecretKey = secretKey;
+			if (secretKey.length === LEGACY_PRIVATE_KEY_SIZE) {
+				// This is a legacy secret key, we need to strip the public key bytes and only read the first 32 bytes
+				pureSecretKey = secretKey.slice(0, PRIVATE_KEY_SIZE);
+			}
+			return Ed25519Keypair.fromSecretKey(pureSecretKey);
```

Comment:
Importing these classes into a file that is imported by the keypairs creates an import cycle which we try to avoid in the SDK.  This probably wouldn't cause issues in most cases, but is often a sign things should be organized a little differently.

@Jordan-Mysten we have tried to keep the number of imports that result in loading all the KeyPair implementations to a minumum (currently it's `@mysten/sui/multisig` and `@mysten/sui/verify`).  It is a bit inconvenient, and as long as we don't reference these methods in other methods they can be removed via tree-shaking.  Curious about your more recent thoughts on this.   We could move this to something `@mysten/sui/keypairs/utils` or something where we have all the methods that require all the keypairs.

Reply:
Personally, I've just avoided adding some of these types of methods because figuring out how to fit them into our exports was too annoying and just left it to the user to implement, which isn't ideal either.

Reply:
Yeah I think we should probably shove this into utils and warn users about the implications of using it (e.g. only use it if you really need to, consider implementing yourself).

We could also revisit DI approaches to solving this.

Reply:
Since we probably want to get this out, lets not try to figure out the DI thing yet.

You can add a new export (keypairs/utils) in the package.json exports https://github.com/MystenLabs/sui/blob/e0ebdd23b9bf4eeb9b6e48569e6c5f8403d14b91/sdk/typescript/package.json#L56-L59 and create a new utils file inside keypairs directory with this util function.  You'll also need to run `pnpm build` from sdk/typescript to generate a few files needed for the export to work correctly in all environments

Reply:
@joyqvq feel free to ping me if you have any questions or issues getting this set up

Reply:
i came up with a way to define a new type `ParsedKeypair` and can void dealing with the concrete keypair, ptal